docs/guides/integration/pytorch.md

```diff
@@ -0,0 +1,167 @@
+    name = "torch-cu121"
+    url = "https://download.pytorch.org/whl/cu121"
+    explicit = true
+    ```
+
+=== "CUDA 12.4"
+
+    ```toml
+    [[tool.uv.index]]
+    name = "torch-cu124"
+    url = "https://download.pytorch.org/whl/cu124"
+    explicit = true
+    ```
+
+=== "ROCm6"
+
+    ```toml
+    [[tool.uv.index]]
+    name = "torch-rocm"
+    url = "https://download.pytorch.org/whl/rocm6.2"
+    explicit = true
+    ```
+
+Note that we also specify the `explicit` option: this prevents packages from being installed from that index unless explicitly pinned to it (see the step below). This means that only PyTorch will be installed from this index, while all other packages will be looked up on PyPI.
+
+## Pin PyTorch to the custom index
+
+Now we need to pin specific PyTorch versions to the appropriate indexes. We do this by adding/editing the `sources` section in the `pyproject.toml`.
+
+=== "CPU-only"
+
+    Note that we use the `platform_system` marker to instruct `uv` to look into this index on Linux and Windows.
+
+    ```toml
+    [tool.uv.sources]
+    torch = [
+      { index = "torch-cpu", marker = "platform_system != 'Darwin'"},
+    ]
+    ```
+
+=== "CUDA 11.8"
+
+    Note that we use the `platform_system` marker to instruct `uv` to look into this index on Linux and Windows.
+
+    ```toml
+    [tool.uv.sources]
+    torch = [
+      { index = "torch-cu118", marker = "platform_system != 'Darwin'"},
+    ]
+    ```
+
+=== "CUDA 12.1"
+
+    Note that we use the `platform_system` marker to instruct `uv` to look into this index on Linux and Windows.
+
+    ```toml
+    [tool.uv.sources]
+    torch = [
+      { index = "torch-cu121", marker = "platform_system != 'Darwin'"},
+    ]
+    ```
+
+=== "CUDA 12.4"
+
+    Note that we use the `platform_system` marker to instruct `uv` to look into this index on Linux and Windows.
+
+    ```toml
+    [tool.uv.sources]
+    torch = [
+      { index = "torch-cu124", marker = "platform_system != 'Darwin'"},
+    ]
+    ```
+
+=== "ROCm6"
+
+    Note that we use the `platform_system` marker to instruct `uv` to look into this index when on Linux only.
+
+    ```toml
+    [tool.uv.sources]
+    torch = [
+      { index = "torch-rocm", marker = "platform_system == 'Linux'"},
+    ]
+    ```
+
+## Add PyTorch to your dependencies
+
+Finally, we can add PyTorch to the `project.dependencies` section of the `pyproject.toml`. You can do this by hand, or using `uv`:
+
+```sh
+uv add torch
+```
+
+However, if you want to be more explicit, you could also:
+
+```toml
+[project]
+dependencies = [
+  "torch==2.5.0 ; platform_system == 'Darwin'",
+  "torch==2.5.0+cu124 ; platform_system != 'Darwin'",
+]
+```
+
+This will install PyTorch 2.5.0 on macOS, and PyTorch 2.5.0+cu124 on Linux and Windows.
+
+!!! warning "PyTorch on Intel Macs"
+
+    Note that the last version to support Intel Macs is PyTorch 2.3.0. In other words, if you try to install PyTorch 2.4.0 or more on an Intel Mac, you will get an error.
```

Comment:
Looks great! Just a nit: seems it is 2.2.x that is supported and 2.3.0 does not support (if I understand correctly)

Reply:
While the new PyTorch version isn't available on PyPI or PyTorch Index for macOS Intel, you can still get it by installing from conda.
![image](https://github.com/user-attachments/assets/d2823f47-f132-4f47-827c-ddcf32986f4f)

Reply:
> Just a nit: seems it is 2.2.x that is supported and 2.3.0 does not support

You are correct! Thanks for spotting this

> you can still get it by installing from conda.

Interesting. I'll edit accordingly. Do we need to link to conda, and if so, what's the most useful link?

Reply:
> Interesting. I'll edit accordingly. Do we need to link to conda, and if so, what's the most useful link?

@baggiponte  
The official [pytorch::pytorch](https://anaconda.org/pytorch/pytorch) (conda) is only built for macOS Intel up to version 2.2.2.
![image](https://github.com/user-attachments/assets/b4a552a9-cac3-477a-91ba-04558b4ab6c9)

The current support is actually community-supported, so I'm uncertain about its quality.
https://github.com/conda-forge/pytorch-cpu-feedstock/blob/main/README.md#current-release-info
![image](https://github.com/user-attachments/assets/e44888e0-1578-4c31-b40c-8fb253dfa03e)


For this reason, I'm not sure if it's advisable to include it in the document.
 Or it should be mentioned that it is community-supported, rather than giving readers the impression that it is officially supported.